packages/jsActions/native-mobile-resources/src/camera/TakePicture.ts

```diff
@@ -50,89 +51,160 @@ export async function TakePicture(
-            }
-            return storeFile(picture, uri);
-        })
-        .catch(error => {
-            if (error === "canceled") {
-                return false;
-            }
-            throw new Error(error);
-        });
+    // V3 dropped the feature of providing an action sheet so users can decide on which action to take, camera or library.
+    const nativeVersionMajor = NativeModules.ImagePickerManager.showImagePicker ? 2 : 4;
+    const RNPermissions = nativeVersionMajor === 4 ? require("react-native-permissions") : null;
+
+    try {
+        const uri = await takePicture();
+
+        if (!uri) {
+            return false;
+        }
+
+        return storeFile(picture, uri);
+    } catch (error) {
+        if (error === "canceled") {
+            return false;
+        }
+
+        throw new Error(error);
+    }
 
     function takePicture(): Promise<string | undefined> {
         return new Promise((resolve, reject) => {
-            const options = getOptions();
-            const method = getPictureMethod();
+            const options = nativeVersionMajor === 2 ? getOptionsV2() : getOptionsV4();
+            getPictureMethod()
+                .then(method =>
+                    method(options, (response: ImagePickerV2Response | ImagePickerResponse) => {
+                        if (response.didCancel) {
+                            return resolve(undefined);
+                        }
 
-            method(options, response => {
-                if (response.didCancel) {
-                    return resolve();
-                }
+                        if (nativeVersionMajor === 2) {
+                            response = response as ImagePickerV2Response;
 
-                if (response.error) {
-                    const unhandledError = handleImagePickerError(response.error);
-                    if (!unhandledError) {
-                        return resolve();
-                    }
-                    return reject(new Error(response.error));
-                }
+                            if (response.error) {
+                                const unhandledError = handleImagePickerV2Error(response.error);
+
+                                if (!unhandledError) {
+                                    return resolve(undefined);
+                                }
+
+                                return reject(new Error(response.error));
+                            }
+
+                            return resolve(response.uri);
+                        }
+
+                        response = response as ImagePickerResponse;
+
+                        if (response.errorCode) {
+                            handleImagePickerV4Error(response.errorCode, response.errorMessage);
+
+                            return resolve(undefined);
+                        }
 
-                return resolve(response.uri);
-            });
+                        return resolve(response.assets[0].uri);
```

Comment:
Unsafe method.. better check for assets array before

Reply:
I don't think it's necessary; this line is reached when there are no library errors, meaning there will be assets. Anything to the contrary would be highly exceptional and cause an error.